purchases/src/test/java/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -3424,166 +3712,90 @@ class PurchasesTest {
+    @Test
+    fun `on billing wrapper connected, query purchases`() {
+        mockSuccessfulQueryPurchases(
+            queriedSUBS = emptyMap(),
+            queriedINAPP = emptyMap(),
+            notInCache = emptyList()
         )
-
-        verify(exactly = 0) {
-            mockBillingAbstract.makePurchaseAsync(
-                eq(mockActivity),
-                eq(appUserId),
-                storeProduct,
-                storeProduct.purchaseOptions[0],
-                ReplaceSkuInfo(oldPurchase),
-                stubOfferingIdentifier
-            )
+        capturedBillingWrapperStateListener.captured.onConnected()
+        verify(exactly = 1) {
+            mockBillingAbstract.queryPurchases(appUserId, any(), any())
         }
-        assertThat(receivedError).isNotNull
-        assertThat(receivedError!!.code).isEqualTo(PurchasesErrorCode.StoreProblemError)
-        assertThat(receivedUserCancelled).isFalse()
     }
 
     @Test
-    fun `Deferred downgrade`() {
-        val (storeProduct, offerings) = stubOfferings("onemonth_freetrial")
-        val oldProductId = "oldProductId"
-
-        val oldPurchase = mockk<StoreTransaction>()
-        every { oldPurchase.productIds[0] } returns oldProductId
-        every { oldPurchase.type } returns ProductType.SUBS
-
-        every {
-            mockBillingAbstract.findPurchaseInPurchaseHistory(
-                appUserId,
-                ProductType.SUBS,
-                oldProductId,
-                captureLambda(),
-                any()
-            )
-        } answers {
-            lambda<(StoreTransaction) -> Unit>().captured.invoke(oldPurchase)
+    fun `on app foregrounded query purchases`() {
+        mockSuccessfulQueryPurchases(
+            queriedSUBS = emptyMap(),
+            queriedINAPP = emptyMap(),
+            notInCache = emptyList()
+        )
+        mockSynchronizeSubscriberAttributesForAllUsers()
+        purchases.onAppForegrounded()
+        verify(exactly = 1) {
+            mockBillingAbstract.queryPurchases(appUserId, any(), any())
         }
+    }
 
-        purchases.purchasePackageOptionWith(
-            mockActivity,
-            offerings[stubOfferingIdentifier]!!.monthly!!,
-            storeProduct.purchaseOptions[0],
-            UpgradeInfo(oldPurchase.productIds[0])
-        ) { _, _ -> }
-
-        capturedPurchasesUpdatedListener.captured.onPurchasesUpdated(emptyList())
+    @Test
+    fun `if billing client not connected do not query purchases`() {
+        every {
+            mockBillingAbstract.isConnected()
+        } returns false
+        purchases.updatePendingPurchaseQueue()
+        verify(exactly = 0) {
+            mockBillingAbstract.queryPurchases(ProductType.SUBS.toGoogleProductType()!!, any(), any())
+        }
+        verify(exactly = 0) {
+            mockBillingAbstract.queryPurchases(ProductType.INAPP.toGoogleProductType()!!, any(), any())
+        }
     }
 
+    // endregion
+
     // region Private Methods
```

Comment:
Looks like we still have a few tests in this region. Would be good to move them to a different place. 
<img width="507" alt="image" src="https://user-images.githubusercontent.com/808417/206160667-952573bf-4e63-47be-b40d-6fde2a23acdc.png">

Also, it might be good to move all the mocking methods somewhere but that shouldn't block this PR.

Reply:
oooooops, yes let me check that out. where did you find this region view though!?

Reply:
It's to the left of android studio, the "structure" view. It's nice to get a better view of big classes 
<img width="321" alt="image" src="https://user-images.githubusercontent.com/808417/206515455-66bf715c-00e2-451f-9bca-6a53cd23bc26.png">